File: backend/schemas/employee.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime

class EmployeeBase(BaseModel):
    employee_code: str
    full_name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    department: Optional[str] = None
    role: Optional[str] = None
    salary: Optional[float] = None

class EmployeeCreate(EmployeeBase):
    pass

class EmployeeUpdate(BaseModel):
    full_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    department: Optional[str] = None
    role: Optional[str] = None
    status: Optional[str] = None
    salary: Optional[float] = None

class EmployeeResponse(EmployeeBase):
    id: int
    status: str
    consent_given_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/employee.ts
export interface Employee {
  id: number;
  employee_code: string;
  full_name: string;
  email?: string;
  phone?: string;
  department?: string;
  role?: string;
  status: 'active' | 'inactive';
  salary?: number;
  consent_given_at?: string;
  created_at: string;
  updated_at: string;
}
